src/main/java/frc/robot/systems/ElevatorArmFSM.java

```diff
@@ -0,0 +1,298 @@
+	 * specific handlers.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 */
+	public void update(TeleopInput input) {
+		if (input == null) {
+			return;
+		}
+		if (currentState != FSMState.IDLE) {
+			currentEncoder = armMotor.getEncoder().getPosition();
+		}
+		if (limitSwitchLow.isPressed()) {
+			armMotor.getEncoder().setPosition(0);
+			currentEncoder = 0;
+		}
+		switch (currentState) {
+			case IDLE:
+				handleIdleState(input);
+				break;
+			case HIGH:
+				handleHighState(input);
+				break;
+			case MIDDLE:
+				handleMiddleState(input);
+				break;
+			case LOW:
+				handleLowState(input);
+				break;
+			case MOVING:
+				handleMovingState(input);
+				break;
+			case ZEROING:
+				handleZeroingState(input);
+				break;
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+		SmartDashboard.putString("Current State", currentState.toString());
+		SmartDashboard.putNumber("Elevator Encoder", armMotor.getEncoder().getPosition());
+		SmartDashboard.putNumber("Elevator Power", armMotor.getAppliedOutput());
+		currentState = nextState(input);
+	}
+
+	/**
+	 * Autonomous update.
+	 * @param autonState current autonomous state
+	 * @return status update
+	 */
+	public boolean autonomousUpdate(FSMState autonState) {
+		if (limitSwitchLow.isPressed()) {
+			armMotor.getEncoder().setPosition(0);
+		}
+
+		switch (autonState) {
+			case IDLE:
+				handleAutonIdleState();
+				return false;
+			case HIGH:
+				handleAutonHighState();
+				return false;
+			case MIDDLE:
+				handleAutonMiddleState();
+				return false;
+			case LOW:
+				handleAutonLowState();
+				return false;
+			case MOVING:
+				return false;
+			case ZEROING:
+				return false;
+			default:
+				return true;
+		}
+	}
+
+	/* ======================== Private methods ======================== */
+	/**
+	 * Decide the next state to transition to. This is a function of the inputs
+	 * and the current state of this FSM. This method should not have any side
+	 * effects on outputs. In other words, this method should only read or get
+	 * values to decide what state to go to.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 * @return FSM state for the next iteration
+	 */
+	private FSMState nextState(TeleopInput input) {
+		if (input == null) {
+			return FSMState.IDLE;
+		}
+		switch (currentState) {
+			case IDLE:
+				if (input.isHighButtonPressed()) {
+					return FSMState.HIGH;
+				} else if (input.isMidButtonPressed()) {
+					return FSMState.MIDDLE;
+				} else if (input.isLowButtonPressed()) {
+					return FSMState.LOW;
+				} else if (Math.abs(input.getLeftJoystickY()) > JOYSTICK_DRIFT_THRESHOLD) {
+					return FSMState.MOVING;
+				} else if (input.isArmZeroButtonPressed()) {
+					return FSMState.ZEROING;
+				}
```

Comment:
You should put a final "else" to return something if none of these conditions are true. Or it's going to drop through into the next case block.